src/main/kotlin/com/exactpro/th2/check1/rule/sequence/SequenceCheckRuleTask.kt

```diff
@@ -57,51 +58,64 @@ import kotlin.collections.set
  * If **checkOrder** parameter is set to `true` the messages must be received in the exact same order as filters were specified.
  * If this parameter is set to `false`, the order won't be checked.
  */
+
 class SequenceCheckRuleTask(
     ruleConfiguration: RuleConfiguration,
     startTime: Instant,
     sessionKey: SessionKey,
     protoPreFilter: PreFilter,
-    private val protoMessageFilters: List<RootMessageFilter>,
+    protoMessageFilters: List<RootMessageFilter>,
     private val checkOrder: Boolean,
     parentEventID: EventID,
     messageStream: Observable<StreamContainer>,
     eventBatchRouter: MessageRouter<EventBatch>
 ) : AbstractCheckTask(ruleConfiguration, startTime, sessionKey, parentEventID, messageStream, eventBatchRouter) {
 
-    private val protoPreMessageFilter: RootMessageFilter = protoPreFilter.toRootMessageFilter()
-    private val messagePreFilter = SailfishFilter(
-        CONVERTER.fromProtoPreFilter(protoPreMessageFilter),
-        protoPreMessageFilter.toCompareSettings()
-    )
-    private val metadataPreFilter: SailfishFilter? = protoPreMessageFilter.metadataFilterOrNull()?.let {
-            SailfishFilter(
-                CONVERTER.fromMetadataFilter(it, VerificationUtil.METADATA_MESSAGE_NAME),
-                it.toComparisonSettings()
-            )
-    }
-    private lateinit var preFilteringResults: MutableMap<MessageID, ComparisonContainer>
-
-    /**
-     * List of filters which haven't matched yet. It is created from the requested filters and reduced after every match
-     */
-    private lateinit var messageFilters: MutableList<MessageFilterContainer>
+    protected class Refs(
+        val protoMessageFilters: List<RootMessageFilter>,
+        val protoPreMessageFilter: RootMessageFilter,
+        val messagePreFilter: SailfishFilter,
+        val metadataPreFilter: SailfishFilter?,
+    ) {
+        val preFilterEvent: Event by lazy {
+            Event.start()
+                .type("preFiltering")
+                .bodyData(protoPreMessageFilter.toReadableBodyCollection())
+        }
 
-    private lateinit var messageFilteringResults: MutableMap<MessageID, ComparisonContainer>
+        val preFilteringResults: MutableMap<MessageID, ComparisonContainer> by lazy { LinkedHashMap() }
```

Comment:
I think we can make it `val` because the `lazy` will create some objects as well. So, I don't think that using lazy for light objects is better that simple reference to the object